src/Translatable/Query/TreeWalker/TranslationWalker.php

```diff
@@ -457,8 +458,13 @@ private function getCastedForeignKey($component, $typeFK, $typePK)
             return $component;
         }
 
+        /*
+         * doctrine/dbal 3.x drops support for PostgreSQL 9.3 and earlier,
+         * the below check prefers the `PostgreSQL94Platform` class as it is available in both major DBAL versions
+         * and falls back to the deprecated `PostgreSqlPlatform` class to account for older PostgreSQL versions with DBAL 2.x
+         */
         // try to look at postgres casting
```

Comment:
IMO, we should avoid block comments inside a method's body.
```suggestion
        // doctrine/dbal 3.x drops support for PostgreSQL 9.3 and earlier, the below check prefers the 
        // `PostgreSQL94Platform` class as it is available in both major DBAL versions and falls back to the 
        // deprecated `PostgreSqlPlatform` class to account for older PostgreSQL versions with DBAL 2.x.
        // Try to look at PostgreSQL casting.
```